hazelcast/src/main/java/com/hazelcast/replicatedmap/merge/ReplicatedMapMergePolicy.java

```diff
@@ -21,25 +21,27 @@
 import java.io.Serializable;
 
 /**
- * A policy for merging replicated maps after a split-brain was detected and the different network partitions need
- * to be merged.
+ * Policy for merging replicated map entries after a split-brain has been healed.
  *
  * @see com.hazelcast.replicatedmap.merge.HigherHitsMapMergePolicy
- * @see com.hazelcast.replicatedmap.merge.PutIfAbsentMapMergePolicy
  * @see com.hazelcast.replicatedmap.merge.LatestUpdateMapMergePolicy
  * @see com.hazelcast.replicatedmap.merge.PassThroughMergePolicy
+ * @see com.hazelcast.replicatedmap.merge.PutIfAbsentMapMergePolicy
  */
 public interface ReplicatedMapMergePolicy extends Serializable {
 
     /**
-     * Returns the value of the entry after the merge
-     * of entries with the same key.
-     * You should consider the case where existingEntry's value is null.
+     * Selects one of the merging and existing map entries to be merged.
+     * <p>
+     * Note that as mentioned also in arguments, the {@link ReplicatedMapEntryView} instance that represents
```

Comment:
It took me some time to wrap my head around this 
> if there is no existing entry for the specified key in the ReplicatedMapEntryView instance that represents the merging map entry 

How about using something like 
> if there is no existing entry with the same key as the key of the merging map entry

This comment applies to the other interfaces as well.

Reply:
> Note that the {@code existingEntry} may be {@code null} if there is no entry with the same key in the destination map.
> This happens, when the entry for that key was only created in the smaller sub-cluster during the split-brain.

I hope this is more understandable (and still correct).

Reply:
It's clearer but there is this issue that if you get a null value you can't be sure if it existed and was deleted or didn't exist and was created. See: https://github.com/hazelcast/hazelcast/issues/10461

Reply:
Oh my, I don't think this can be easily fixed. Maybe you can do some black magic with the creation time of the merging entry and try to figure out when the split brain was detected (do we have a listener for this?).

So I hope this is the best way to document this for now:
```java
/**
 * Selects one of the merging and existing map entries to be merged.
 * <p>
 * Note that the {@code existingEntry} may be {@code null} if there is no entry
 * with the same key in the destination map.
 * This happens, when the entry for that key was
 * <ul>
 * <li>only created in the smaller sub-cluster during the split-brain</li>
 * <li>removed in the larger sub-cluster during the split-brain</li>
 * </ul>
 */
```

Reply:
Looks good. I would be very wary of using timestamps for correctness since they can be known for their flakiness. There might be a way but I don't see it.